scripts/docker/update_dockers_json.py

```diff
@@ -0,0 +1,212 @@
+                    "tag.\n\n"
+                    "If the given tag is different from the tag in the "
+                    "provided JSON file, and an image with the given tag "
+                    "exists in the registry, the script updates the "
+                    "referenced Docker image; otherwise, it leaves the "
+                    "referenced image unchanged. The script persists the "
+                    "updated list of images in the provided output JSON file."
+                    "\n\n"
+                    "The intended use-case of this script is for a scenario "
+                    "when you know the tag used to build and push GATK-SV "
+                    "Docker images using the `build_docker.py` script, "
+                    "however, you're not sure which images are updated "
+                    "(since `build_docker.py` may rebuild and push additional"
+                    "images to the specified images, e.g., its dependencies). "
+                    "This script is predominantly developed for the Github "
+                    "actions, and is a temporary solution until the "
+                    "`build_docker.py` is re-written to provide this "
+                    "functionality.",
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument(
+        "dockers_json",
+        help="A JSON file containing Docker images to check. ")
+
+    parser.add_argument(
+        "image_tag",
+        help="Docker image tag.")
+
+    parser.add_argument(
+        "-u", "--updated_dockers_json",
+        help="[Optional] A JSON file to persist a list of the "
+             "updated Docker images. Defaults to a filename as the "
+             "input `dockers_json`. Note that if the given file exists, "
+             "this script will replace it."
+    )
+
+    args = parser.parse_args()
+
+    # Checks if the file exists, if not, errors-out.
+    os.path.isfile(args.dockers_json)
+
+    # Loads the JSON object from `args.dockers_json`.
+    # Errors-out if an invalid JSON is provide.
+    with open(args.dockers_json, "r") as f:
+        images = json.load(f)
+
+    updated_dockers_json = args.updated_dockers_json \
+        if args.updated_dockers_json else args.dockers_json
+
+    # This ensures that the script can write to the path
+    # of the provided JSON file; hence can early break
+    # before running all the Docker assertions if it does
+    # not have write access to the output path.
+    path_parts = Path(updated_dockers_json).parts
+    tmp = Path(*path_parts[:-1], f"tmp_{path_parts[-1]}")
+    # or the following; however, pathlib.Path is more portable
+    # than os.path and is a newer alternative.
+    # tmp = os.path.join(os.path.dirname(updated_dockers_json), "tmp")
+    with open(tmp, "w+") as f:
+        json.dump({}, f)
+    os.remove(tmp)
+
+    updated_images = get_updated_images(images, args.image_tag)
+    with open(updated_dockers_json, "w+") as f:
```

Comment:
I don't understand why this is opened `w+` (reading and writing) since it will not be read from.
```suggestion
    with open(updated_dockers_json, "w") as f:
```